### PR TITLE
[typescript-axios] [Java] Respect User Supplied Header Params

### PR DESCRIPTION
**Pull request by ckoegel**
This PR moves the assignment of user supplied header parameters after the automatic OAS defined assignment. This allows the user to define a header parameter and have it be respected instead of overwritten. Changes were made to both the Java `okhttp-gson` and `typescript-axios` generators to apply this change in both.

### PR checklist
 
- [x] Read the [contribution guidelines](https://github.com/openapitools/openapi-generator/blob/master/CONTRIBUTING.md).
- [x] Pull Request title clearly describes the work in the pull request and Pull Request description provides details about how to validate the work. Missing information here may result in delayed response from the community.
- [x] Run the following to [build the project](https://github.com/OpenAPITools/openapi-generator#14---build-projects) and update samples:
  ```
  ./mvnw clean package 
  ./bin/generate-samples.sh ./bin/configs/*.yaml
  ./bin/utils/export_docs_generators.sh
  ``` 
  (For Windows users, please run the script in [Git BASH](https://gitforwindows.org/))
  Commit all changed files. 
  This is important, as CI jobs will verify _all_ generator outputs of your HEAD commit as it would merge with master. 
  These must match the expectations made by your contribution. 
  You may regenerate an individual generator by passing the relevant config(s) as an argument to the script, for example `./bin/generate-samples.sh bin/configs/java*`. 
  IMPORTANT: Do **NOT** purge/delete any folders/files (e.g. tests) when regenerating the samples as manually written tests may be removed.
- [x] File the PR against the [correct branch](https://github.com/OpenAPITools/openapi-generator/wiki/Git-Branches): `master` (upcoming `7.x.0` minor release - breaking changes with fallbacks), `8.0.x` (breaking changes without fallbacks)
- [x] If your PR is targeting a particular programming language, @mention the [technical committee](https://github.com/openapitools/openapi-generator/#62---openapi-generator-technical-committee) members, so they are more likely to review the pull request.

Java - @bbdouglas (2017/07) @sreeshas (2017/08) @jfiala (2017/08) @lukoyanov (2017/09) @cbornet (2017/09) @jeff9finger (2018/01) @karismann (2019/03) @Zomzog (2019/04) @lwlee2608 (2019/10) @martin-mfg (2023/08)

TypeScript - @TiFu (2017/07) @taxpon (2017/07) @sebastianhaas (2017/07) @kenisteward (2017/07) @Vrolijkx (2017/09) @macjohnny (2018/01) @topce (2018/10) @akehir (2019/07) @petejohansonxo (2019/11) @amakhrov (2020/02) @davidgamero (2022/03) @mkusaka (2022/04) @joscha (2024/10)

### CODE REVIEW

**Comment by joscha** [comment]
Intent lgtm, unsure about implications, so let's wait for @macjohnny to weigh in.

File: samples/client/petstore/typescript-axios/tests/with-complex-headers/pom.xml [hunk 1]
```diff
@@ -0,0 +1,59 @@
+<project>
```
Comment (macjohnny) [inline samples/client/petstore/typescript-axios/tests/with-complex-headers/pom.xml:1]: this has to be registered similar to https://github.com/OpenAPITools/openapi-generator/blob/57cfff195080e4f983d849d41037584132f326ee/CI/circle_parallel.sh#L101

**Review by macjohnny — APPROVED** [review]
Thanks for your contribution!

**Comment by macjohnny** [comment]
@ckoegel please split this PR into two, one for the java generator and one for the typescript generator, so it is easier to revert in case things go wrong

**Comment by ckoegel** [comment]
@macjohnny 
Closing this in favor of #20040 and #20042 which split this PR based on language

**Pull request closed without merging**